design/baremetal-operator/bmh_live_boot.md

```diff
@@ -0,0 +1,121 @@
+for example the [fedora-coreos installer](https://docs.fedoraproject.org/en-US/fedora-coreos/bare-metal/)
+
+An [Ironic API](https://docs.openstack.org/ironic/latest/admin/ramdisk-boot.html)
+exists that supports this but it's not currently accessible via metal3.
+
+### Goals
+
+Expose the Ironic API to boot iso images via metal3.
+
+### Non-Goals
+
+This only considers the ability to boot an iso image (which Ironic supports
+via both redfish-virtualmedia and iPXE), not live kernel/ramdisk images.
+
+Ironic does support booting kernel/ramdisk images though, so the interface
+we decide on shouldn't preclude adding that capability in future.
+
+## Proposal
+
+Add a new field to the BareMetalHost spec which indicates that instead of
+deploying an image to disk, a live-iso image will be booted:
+
+```yaml
+  spec:
+    liveImage:
+      url: http://1.2.3.4/image.iso
+    online: true
+```
+
+Note that in this mode, `rootDeviceHints` and `userData` will not be used
+since the image won't be written to disk, and Ironic doesn't
+support passing user-data in addition to the iso attachment at present.
+
+At some future point it would be desirable for Ironic to add the ability
+to pass userData - then a generic iso could be booted with the ability
+to define customization via the userData field in the normal way.
+
+## Design Details
+
+When this mode is selected, we need to configure Ironic to use the [ramdisk
+deploy interface](https://docs.openstack.org/ironic/latest/admin/ramdisk-boot.html)
+which is not currently enabled in ironic-image.
+
+We will also need to write the provided ISO URL into the
+[instance_info boot_iso field](https://docs.openstack.org/ironic/latest/admin/drivers/redfish.html#virtual-media-ramdisk)
+
+### Risks and Mitigations
+
+- Currently there is no detachment API in ironic, so in the case where an installer
+  iso is booted, we don't have a way to detach the virtualmedia and will have to
+  rely e.g on efibootmgr in whatever iso gets booted to ensure the correct boot device.
+
+- The inspection of a BMH will still rely on booting IPA and there's no support for
+  fast-track provisioning in this workflow (since it's booting two different
+  iso images) so this adds a reboot into the process.  This can potentially be
+  avoided in some cases where we can provide a status annotation on creation of
+  the BMH such that inspection is not performed.
+
+- Similarly cleaning of a BMH will rely on IPA, so will require a reboot which may
+  not be desirable in some situations.  This can potentially be avoided if cleaning
+  is [made configurable](https://github.com/metal3-io/metal3-docs/pull/151) in future.
+
+### Work Items
+
+- Enable ramdisk deploy interface in ironic-image
+- Add `iso` field to the BMH schema
+- Update BMO deploy logic to set the `deploy_interface` and `instance_info` appropriately
+
+### Dependencies
+
+TODO link Ironic roadmap items as soft dependencies
```

Comment:
Is this still a TODO?

Reply:
Yes I've asked for some links from @dtantsur in the comments above, we can merge without if needed though and I'll add them as a follow-up

Reply:
Yeah, I'm trying to help us keep moving. These documents are meant to be kept up to date, so I think it's OK to add links here later if we know there is work planned but don't have artifacts for that work, yet.

Reply:
The links have been provided, see above.

Reply:
I added a new commit with the links, thanks!